### PR TITLE
tools/unit-tests-node

### PR DESCRIPTION
**Pull request by goransle**
Changed the tests in the `ts-node-unit-test` folder to use the now stable Node test runner. 

The tests now feature a more standard `describe` and `it` syntax, and the reporter should be more human readable. (Also the tests should be more transportable if we decide to move to another test runner at some point)

They can now be run with
* the new `npm run test-node` script
* or directly (depending on your Node version and OS) `npx tsx --test ./test/ts-node-unit-tests/tests/**` (or `node --import tsx --test ./test/ts-node-unit-tests/tests/**`)

Seems to require Node 20 (lts) for errors to be reported, but also works in Node 18.

### CODE REVIEW

**Review by KacperMadej — CHANGES_REQUESTED** [review]
On Windows this doesn't run when using PowerShell nor the standard cmd. It runs ok on `git bash` and 61 tests are passing.

Both `node --import ...` and `npx tsx ...` gives `"Could not find 'C:\path\**' "` (with the correct path as given in the argument). When the path is given without any `*` it runs but finds 0 tests to run like:
```
npx tsx --test ./test/ts-node-unit-tests/tests/  
ℹ tests 0
ℹ suites 0
ℹ pass 0
ℹ fail 0
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 7.5203
```

**Comment by goransle** [comment]
@KacperMadej Hopefully it should now be runnable using `npm run test-node` 🤞

File: test/ts-node-unit-tests/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import * as glob from 'glob';
+const files = glob.sync(join(__dirname, 'tests') + '/**/*.test.ts');
+
+// Workaround file for `node --test` not working with Windows and Node 20
+// TODO: consider removing when Node 22 is LTS
+
+run({
+    files,
+    watch: process.argv.includes('--watch')
+}).on('test:fail', () => {
```
Comment (KacperMadej) [inline test/ts-node-unit-tests/index.ts:14]: I don't get anything in the terminal on a fail (I changed a test to get a fail). Maybe something like `}).on('test:fail', (e) => {
    console.error('Test failed: ', e);` or equivalent could be added?

**Review by KacperMadej — COMMENTED** [review]
It is running! 🎉

File: test/ts-node-unit-tests/index.ts (hunk 1 above)
Comment (goransle) [reply test/ts-node-unit-tests/index.ts:14]: 🤔 hmm, the reporter gives me this on Windows (running in paralells) with Node 20: 
```sh
ℹ tests 61
ℹ suites 12
ℹ pass 60
ℹ fail 1
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 6669.1485

✖ failing tests:

test at test\ts-node-unit-tests\tests\global-utils.test.ts:2:1255
✖ correctly formats a date (6.924042ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected

  + 'Sunday,  1 Jan, 00:00:00'
  - 'Sunday,  2 Jan, 00:00:00'
              ^
      at TestContext.<anonymous> (C:\Users\goranslettemark\highcharts\test\ts-node-unit-tests\tests\global-utils.test.ts:72:13)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:789:25)
      at Test.start (node:internal/test_runner/test:693:17)
      at node:internal/test_runner/test:1129:71
      at node:internal/per_context/primordials:487:82
      at new Promise (<anonymous>)
      at new SafePromise (node:internal/per_context/primordials:455:29)
      at node:internal/per_context/primordials:487:9
      at Array.map (<anonymous>) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'Sunday,  1 Jan, 00:00:00',
    expected: 'Sunday,  2 Jan, 00:00:00',
    operator: 'strictEqual'
  }
```

Comment (goransle) [reply test/ts-node-unit-tests/index.ts:14]: What kind of failure did you add?

Comment (KacperMadej) [reply test/ts-node-unit-tests/index.ts:14]: Just some random typo in the expected array in modules.test.ts. I was using Node v18. After updating to v20.17.0 all is perfectly fine.

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**